ibis/backends/base/sql/registry/window.py

```diff
@@ -251,15 +215,18 @@ def window(translator, op):
 
     # Some analytic functions need to have the expression of interest in
     # the ORDER BY part of the window clause
-    if isinstance(arg, _require_order_by) and not window._order_by:
+    if isinstance(arg, translator._require_order_by) and not window._order_by:
         window = window.order_by(arg.args[0])
 
     # Time ranges need to be converted to microseconds.
     # FIXME(kszucs): avoid the expression roundtrip
     if window.how == 'range':
-        order_by_types = [type(x.op().args[0]) for x in window._order_by]
-        time_range_types = (ir.TimeColumn, ir.DateColumn, ir.TimestampColumn)
-        if any(col_type in time_range_types for col_type in order_by_types):
+        time_range_types = (dt.Time, dt.Date, dt.Timestamp)
+        if any(
+            isinstance(c.output_dtype, time_range_types)
+            and c.output_shape == rlz.Shape.COLUMNAR
```

Comment:
You should be able to use `c.output_shape.is_columnar()` here.

Reply:
Perhaps a similar `c.output_dtype.is_temporal()` convenience method would be useful too.

Reply:
This was done in #4841.